packages/react-ui/src/lib/data-selector-utils.ts

```diff
@@ -0,0 +1,153 @@
+export type MentionTreeNode = {
+  key: string;
+  data: {
+    propertyPath: string;
+    /**Key for json value */
+    displayName: string;
+    /**value for json key */
+    value?: string | unknown;
+    isSlice: boolean;
+    isTestStepNode?: boolean;
+  };
+  children?: MentionTreeNode[];
+};
+/**Traverses an object to find its child properties and their paths, stepOutput has to be an object on first invocation */
+function traverseStepOutputAndReturnMentionTree({
+  stepOutput,
+  propertyPath,
+  displayName,
+}: {
+  stepOutput: unknown;
+  propertyPath: string;
+  displayName: string;
+}): MentionTreeNode {
+  if (stepOutput && typeof stepOutput === 'object') {
+    if (Array.isArray(stepOutput)) {
+      return handlingArrayStepOutput(stepOutput, propertyPath, displayName);
+    }
```

Comment:
I think javascript is considering array is Object but it's doesn't have to be written like this.

a better way to write this like that, is to move array outside the check of object.

and please extract the stepoutput check to something like this
const isObject = stepOutput && typeof stepOutput === 'object';